taxonomy/views.py

```diff
@@ -0,0 +1,79 @@
+"""
+Views for the taxonomy app
+"""
+
+from django.shortcuts import render, redirect
+
+from .forms import VocabularyForm
+from learningresources.models import Repository
+from taxonomy.models import Vocabulary
+
+
+def edit(request, vocabulary_id=None):
+    """
+    Show form to create a new vocabulary
+    """
+    # TODO: this function can be made simpler
+    if vocabulary_id is None:
+        form = VocabularyForm()
+    else:
+        vocabulary = Vocabulary.objects.get(id=vocabulary_id)
+        form = VocabularyForm(instance=vocabulary)
+
+    # TODO: where to specify this?
+    repository = Repository.objects.first()
+
+    form.instance.repository = repository
+    # TODO: are these good defaults?
+    form.instance.required = False
+    form.instance.weight = 1000
+
+    if request.POST:
```

Comment:
Handling both a GET and a POST in the same view can be handled with a little less code. See this for an example: https://github.com/mitodl/lore/blob/skm_import/importer/views.py#L29